### PR TITLE
[feat]tools-v2: add bs recover volume

### PR DESCRIPTION
**Pull request by CrystalAnalyst**
### What problem does this PR solve?

Issue Number: #2588  

Problem Summary: Add a feature that can recover the specified volume from the RecycleBin.

### What is changed and how it works?

What's Changed: add recover directory, and inside of it is a volume sub-directory, and two files: recover.go, file.go. 

How it Works: recover.go act as an entry, while file.go call the Rpc function to execute the actual recover action.

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [yes] Relevant documentation/comments is changed or added
- [yes] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: tools-v2/pkg/cli/command/curvebs/recover/recover.go
Comment (baytan0720) [inline tools-v2/pkg/cli/command/curvebs/recover/recover.go:1]: This command is a middle command. You can add sub command in middle command. You can refer to tools-v2/pkg/cli/command/curvebs/update/update.go.

File: tools-v2/pkg/cli/command/curvebs/recover/volume/file.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/nameserver2"
+)
+
+type RecoverCertainFileRpc struct {
```
Comment (baytan0720) [inline tools-v2/pkg/cli/command/curvebs/recover/volume/file.go:18]: Named the same as RPC

**Review by baytan0720 — COMMENTED** [review]
Please update the README.md about the new command.
You should post the sample output of the command in description of pr.

File: tools-v2/pkg/cli/command/curvebs/recover/volume/file.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	}
+	return nil
+}
+
+func (recoverCommand *RecoverCommand) AddFlags() {
+	config.AddRpcTimeoutFlag(recoverCommand.Cmd)
+	config.AddRpcRetryTimesFlag(recoverCommand.Cmd)
+	config.AddBsMdsFlagOption(recoverCommand.Cmd)
+	config.AddBsPathRequiredFlag(recoverCommand.Cmd)
+	config.AddBsUserOptionFlag(recoverCommand.Cmd)
```
Comment (baytan0720) [inline tools-v2/pkg/cli/command/curvebs/recover/volume/file.go:81]: this should be required

File: tools-v2/pkg/cli/command/curvebs/recover/volume/file.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	config.AddBsPasswordOptionFlag(recoverCommand.Cmd)
```
Comment (baytan0720) [inline tools-v2/pkg/cli/command/curvebs/recover/volume/file.go:82]: flag `fileid` should be added

**Comment by CrystalAnalyst** [comment]
okay, thanks a lot for your advice, I'll fix it in the next few days.

**Pull request closed without merging**